wondrous-app/components/CreateEntity/CreateEntityModal/TaskTemplatePicker/index.tsx

```diff
@@ -0,0 +1,343 @@
+import { useLazyQuery } from '@apollo/client';
+import { useMe } from 'components/Auth/withAuth';
+import { ActionButton } from 'components/Common/Task/styles';
+import { RichTextViewer } from 'components/RichText';
+import { TaskTemplateActionContainer, TokenGatingTextfieldInput } from 'components/Settings/TokenGating/styles';
+import { GET_TASK_TEMPLATES_BY_USER_ID } from 'graphql/queries';
+import { useEffect, useState } from 'react';
+import {
+  CreateEntityAutocompletePopperRenderInputAdornment,
+  CreateEntityAutocompletePopperRenderInputIcon,
+  CreateEntityTextfieldInputTemplate,
+} from '../styles';
+import {
+  CreateEntityDefaultDaoImage,
+  PodSearchLabel,
+  PodSearchListItem,
+  TaskTemplateActionPopper,
+  TaskTemplateArrowIcon,
+  TaskTemplateBorderLine,
+  TaskTemplateCancelButton,
+  TaskTemplateClickAway,
+  TaskTemplateCloseIcon,
+  TaskTemplateContainer,
+  TaskTemplateDefaultPopper,
+  TaskTemplateDeleteIcon,
+  TaskTemplateEthereumIcon,
+  TaskTemplateIcon,
+  TaskTemplateLabel,
+  TaskTemplateLabelValue,
+  TaskTemplateLabelWrapper,
+  TaskTemplateListItems,
+  TaskTemplateModal,
+  TaskTemplateOptionsLabel,
+  TaskTemplatePointsValue,
+  TaskTemplateRewardBox,
+  TaskTemplateRewardContainer,
+  TaskTemplateRewardValue,
+  TaskTemplateSpecificTitleBar,
+  TaskTemplateTitle,
+  TaskTemplateTitleBar,
+} from './styles';
+import TemplateEllipsesIcon from './TemplateEllipsesIcon';
+
+const TaskTemplatePicker = (props) => {
+  const {
+    options,
+    onChange,
+    value,
+    disabled,
+    handleSubmitTemplate,
+    paymentMethods,
+    handleSaveTemplate,
+    handleEditTemplate,
+    handleDeleteTemplate,
+  } = props;
+  const [anchorEl, setAnchorEl] = useState(null);
+  const [optionAnchorEl, setOptionAnchorEl] = useState(null);
+  const [usedTemplate, setUsedTemplate] = useState(null);
+  const [templateValue, setTemplateValue] = useState('');
+
+  const user = useMe();
+  const handleEllipsesClick = (event) => setOptionAnchorEl(optionAnchorEl ? null : event.currentTarget);
+  const [saveOrOpen, setSaveOrOpen] = useState('none');
+  const handleClick = (event) => setAnchorEl(anchorEl ? null : event.currentTarget);
+  const handleClickAway = () => {
+    setAnchorEl(null);
+    setSaveOrOpen('none');
+  };
+  const open = Boolean(anchorEl);
+  const templateOptionsOpen = Boolean(optionAnchorEl);
+  const [templates, setTemplates] = useState([]);
+
+  const [getTaskTemplates] = useLazyQuery(GET_TASK_TEMPLATES_BY_USER_ID, {
```

Comment:
I think in here if you do `const [getTaskTemplates, {data, loading}] = useLazyQuery(...)` data will contain data?.getTaskTemplatesByUserId and we can remove setTemplates state 
 
then you can refer this in the render like data?.getTaskTemplatesByUserId?.length or you could use the loading and render loading state based on this